webdriver/tests/find_element_from_shadow_root/find.py

```diff
@@ -64,17 +64,15 @@ def test_found_element_equivalence(session, get_shadow_page):
     expected = session.execute_script("return arguments[0].shadowRoot.querySelector('input')",
                                       args=(custom_element,))
     shadow_root = custom_element.shadow_root
-    response = find_element(session, shadow_root.id, "css", "input")
+    response = find_element(session, shadow_root.id, "css selector", "input")
     value = assert_success(response)
     assert_same_element(session, value, expected)
 
 
 @pytest.mark.parametrize("using,value",
                          [("css selector", "#linkText"),
                           ("link text", "full link text"),
-                          ("partial link text", "link text"),
-                          ("tag name", "a"),
-                          ("xpath", "//a")])
```

Comment:
I was testing and `tag name` and `xpath` are not supported in a shadow root element. If you want to, I can add some test to see that an invalid argument is received for those values.

Reply:
For testing I mean testing in `chromium` and `firefox` and they don't support them, but maybe the is just because they don't have them implemented?

Reply:
How exactly did you test? Note that at least Firefox doesn't have a WebDriver implementation for shadow root yet. It's being worked on https://bugzilla.mozilla.org/show_bug.cgi?id=1700073.

Reply:
I was testing javascript. I'm taking also a look to the w3c https://www.w3.org/TR/dom41/ and looks like shadow root doesn't have a method to get by the tag name.

Reply:
you should look at https://dom.spec.whatwg.org/ (sorry, /TR/dom41/ is [updated to redirect](https://github.com/w3c/whatwg-coord/issues/5#issuecomment-895391219)).

Reply:
Thanks @plehegar for the link. I still can't find anything about supporting getting an element by their tag name in a shadow root

Reply:
Maybe I'm interpretating this wrong but, I'm associating the 'tag name' with the method `getElementsByTagName`.

Reply:
I actually asked some developers and it turned out that `getElementsByTagName` returns `HTMLCollection`, which is a legacy thing. So it's probably the reason why it is not supported. Same seems to also apply to `XPath`.

I would propose that you leave in these two selectors for now, and file a new issue for discussion. We may have to remove both selector types for shadow root elements.

Reply:
Actually `xpath` we might be able to keep given that `XPathEvaluator` could always be created, and the `shadowRoot`  can be passed in as start node. That's because `ShadowRoot` is still a `DocumentFragment`.

Reply:
> Actually xpath we might be able to keep given that XPathEvaluator could always be created, and the shadowRoot can be passed in as start node. That's because ShadowRoot is still a DocumentFragment.

Do you have an example? When I test them in the F12 tool (Chromium, Firefox) I get an error. 
In chromium it says `Uncaught DOMException: Failed to execute 'evaluate' on 'XPathExpression': The node provided is '#document-fragment', which is not a valid context node type.`
In firefox it says `Uncaught DOMException: Operation is not supported`

The code I have used is:
```js
const evalutaor = new XPathEvaluator();
const expression = evalutaor.createExpression('//input');
expression.evaluate(shadowroot)
```

> I would propose that you leave in these two selectors for now, and file a new issue for discussion. We may have to remove both selector types for shadow root elements.

I like the idea, so we can unblock this PR. I will update the PR and create the issue for further discussion.

Reply:
I have restored this code and [open an issue](https://github.com/web-platform-tests/wpt/issues/30010) to follow up the discussion :)

Reply:
As already mentioned on the newly filed issue please file an issue for webdriver spec. Then as best copy and paste the above conversation as reference. Thanks!

Reply:
@whimboo the issue in the w3c repo is created and I have added a link to this conversation in the description of the issue.

Reply:
The link to the w3c issue is: https://github.com/w3c/webdriver/issues/1610